src/pybind/mgr/dashboard/controllers/orchestrator.py

```diff
@@ -5,11 +5,33 @@
 
 from . import ApiController, Endpoint, ReadPermission
 from . import RESTController, Task
+from .. import mgr
 from ..security import Scope
 from ..services.orchestrator import OrchClient
 from ..tools import wraps
 
 
+def get_device_osd_map():
+    """Get mappings from inventory devices to OSD IDs.
+
+    :return: Returns a dictionary containing mappings.
+        e.g. {'node1': {'vdc': 2, 'vdb': 1}, 'node2': {'vdc': 0}}
+    :rtype: dict
+    """
+    device_osd_map = {}
+    for osd_id, osd_metadata in mgr.get('osd_metadata').items():
+        hostname = osd_metadata['hostname']
+        devices = osd_metadata.get('devices')
+        if not devices:
+            continue
+        if hostname not in device_osd_map:
+            device_osd_map[hostname] = {}
+        # for OSD contains multiple devices, devices is in `sda,sdb`
+        for device in devices.split(','):
+            device_osd_map[hostname][device] = int(osd_id)
```

Comment:
as discussed, you can have more than one osd per device.

Reply:
Fixed.